pkg/volume/azure_file/azure_file.go

```diff
@@ -260,16 +262,35 @@ func (c *azureFileUnmounter) TearDownAt(dir string) error {
 	return util.UnmountPath(dir, c.mounter)
 }
 
-func getVolumeSource(
-	spec *volume.Spec) (*v1.AzureFileVolumeSource, bool, error) {
+func getVolumeSource(spec *volume.Spec) (string, bool, error) {
 	if spec.Volume != nil && spec.Volume.AzureFile != nil {
-		return spec.Volume.AzureFile, spec.Volume.AzureFile.ReadOnly, nil
+		share := spec.Volume.AzureFile.ShareName
+		readOnly := spec.Volume.AzureFile.ReadOnly
+		return share, readOnly, nil
 	} else if spec.PersistentVolume != nil &&
 		spec.PersistentVolume.Spec.AzureFile != nil {
-		return spec.PersistentVolume.Spec.AzureFile, spec.ReadOnly, nil
+		share := spec.PersistentVolume.Spec.AzureFile.ShareName
+		readOnly := spec.ReadOnly
+		return share, readOnly, nil
 	}
+	return "", false, fmt.Errorf("Spec does not reference an AzureFile volume type")
+}
 
-	return nil, false, fmt.Errorf("Spec does not reference an AzureFile volume type")
+func getSecretNameAndNamespace(spec *volume.Spec, defaultNamespace string) (string, string, error) {
+	if spec.Volume != nil && spec.Volume.AzureFile != nil {
+		secretName := spec.Volume.AzureFile.SecretName
+		secretNamespace := defaultNamespace
+		return secretName, secretNamespace, nil
+	} else if spec.PersistentVolume != nil &&
+		spec.PersistentVolume.Spec.AzureFile != nil {
+		secretNamespace := defaultNamespace
+		if spec.PersistentVolume.Spec.AzureFile.SecretNamespace != nil {
+			secretNamespace = *spec.PersistentVolume.Spec.AzureFile.SecretNamespace
+		}
+		secretName := spec.PersistentVolume.Spec.AzureFile.SecretName
```

Comment:
after determining the namespace (either from defaultNamespace or from spec.PersistentVolume.Spec.AzureFile.SecretNamespace), make sure it is not empty, and return if it is (e.g. `fmt.Errorf("invalid PV spec: nil namespace")`)